.changeset/smart-chefs-fail.md

```diff
@@ -0,0 +1,5 @@
+---
+'@primer/react': minor
+---
+
+IconButton: introduce tooltip behavior behind the prop `_enableTooltip_`
```

Comment:
In this PR, I am only introducing the tooltip behaviour without worrying about the menu anchors. Because this code won't have any impact in the upstream repos and we can address the menu anchor in a seperate PR to keep the PRs small.